### PR TITLE
use add_custom_command for generating protocols

### PR DESCRIPTION
**Pull request by cnt0**
This fixes an issue with build error in case of e.g.

```bash
$ rm protocols/*.h
$ cmake --build build
```

The workaround was to 
```bash
$ touch CMakeLists.txt
``` 
With this PR, protocol files are properly regenerated with no extra efforts.

Also, resolve hyprwayland-scanner dependency via cmake instead of pkg-config.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -109,9 +109,11 @@ pkg_check_modules(deps REQUIRED IMPORTED_TARGET
     wayland-server wayland-client wayland-cursor wayland-protocols
     cairo pango pangocairo pixman-1
     libdrm libinput hwdata libseat libdisplay-info libliftoff libudev gbm
-    hyprwayland-scanner>=0.3.6 hyprlang>=0.3.2 hyprcursor>=0.1.7
+    hyprlang>=0.3.2 hyprcursor>=0.1.7
 )
 
+find_package(hyprwayland-scanner 0.3.6 REQUIRED)
```
Comment (fufexan) [inline CMakeLists.txt:115]: @vaxerski do we need 0.3.7?

Comment (vaxerski) [reply CMakeLists.txt:115]: now we do anyways (see conflict)

Comment (fufexan) [reply CMakeLists.txt:115]: Right. @cnt0 can you update it?

Comment (cnt0) [reply CMakeLists.txt]: fixed & rebased

github seems to show false diffs in this PR, I have no idea why. `CMakeLists.txt` is the only file here that is actually changed compared to the main branch.

Comment (fufexan) [reply CMakeLists.txt]: Caused by merging main into your branch. A rebase should fix it. If it doesn't, `git rebase --interactive HEAD~10` and `delete` any commits that aren't your own.

**Pull request closed without merging**